netmaster/main.go

```diff
@@ -41,148 +43,129 @@ type cliOpts struct {
+		return nil, err
 	}
 
-	if opts.version {
-		fmt.Printf(version.String())
-		os.Exit(0)
+	// 2. validate network configs
+	netConfigs, err := utils.ValidateNetworkOptions(binName, ctx)
+	if err != nil {
+		return nil, err
 	}
 
-	log.SetFormatter(&log.TextFormatter{FullTimestamp: true, TimestampFormat: time.StampNano})
-
-	if opts.debug {
-		log.SetLevel(log.DebugLevel)
+	// 3. validate db configs
+	dbConfigs, err := utils.ValidateDBOptions(binName, ctx)
+	if err != nil {
+		return nil, err
 	}
 
-	// Validate listen and control URL options
-	listenURL := strings.Split(opts.listenURL, ":")
-	controlURL := strings.Split(opts.controlURL, ":")
-	if len(listenURL) != 2 {
-		log.Fatalf("Listen URL not in proper format. Valid format: [IP]:Port")
-	}
-	if len(controlURL) != 2 || (strings.Compare(controlURL[0], "0.0.0.0") == 0) {
-		log.Fatalf("Control URL not in proper format. Valid format: IP:Port")
+	// 4. set v2 plugin name if it's set
+	pluginName := ctx.String("name")
+	if netConfigs.Mode == core.Docker || netConfigs.Mode == core.SwarmMode {
+		logrus.Infof("Using netmaster docker v2 plugin name: %s", pluginName)
+		docknet.UpdateDockerV2PluginName(pluginName, pluginName)
+	} else {
+		logrus.Infof("Ignore netmaster docker v2 plugin name: %s (netmaster mode: %s)", pluginName, netConfigs.Mode)
 	}
 
-	listenIP := listenURL[0]
-	listenPort, err := strconv.Atoi(listenURL[1])
-	if err != nil {
-		log.Fatalf("Listen URL port not in valid format. Err: %+v", err)
+	// 5. set plugin listen addresses
+	externalAddress := ctx.String("external-address")
+	if externalAddress == "" {
+		return nil, errors.New("netmaster external-address is not set")
+	} else if err := netutils.ValidateBindAddress(externalAddress); err != nil {
+		return nil, err
 	}
-	log.Infof("Listen IP:Port %s:%d", listenIP, listenPort)
+	logrus.Infof("Using netmaster external-address: %s", externalAddress)
 
-	controlIP := controlURL[0]
-	controlPort, err := strconv.Atoi(controlURL[1])
-	if err != nil {
-		log.Fatalf("Control URL port not in valid format. Err: %+v", err)
-	}
-	if len(controlIP) == 0 {
-		if strings.Compare(opts.controlURL, defaultControlPort) != 0 {
-			// Case: If --control-url :XXXX, and :XXXX is not defaultControlPort; we error out
-			log.Fatalf("Control URL not in proper format. Valid format: IP:Port")
-		}
-		if len(listenIP) != 0 && (strings.Compare(listenIP, "0.0.0.0") != 0) && (controlPort == listenPort) {
-			// Case: --listen-url A.B.C.D:XXXX and --control-url :XXXX
-			controlIP = listenIP
-		} else {
-			// Case: [--listen-url A.B.C.D:XXXX] --control-url defaultControlPort
-			// Get the address to be used for local communication
-			controlIP, err = daemon.GetLocalAddr()
-			if err != nil {
-				log.Fatalf("Error getting local IP address for Control URL. Err: %v", err)
-			}
-			controlURL[1] = listenURL[1]
+	internalAddress := ctx.String("internal-address")
+	if internalAddress == "" {
+		localIP, err := netutils.GetDefaultAddr()
```

Comment:
it's actually still in help, but not in value, this is better to avoid confusing of different helper in different node, in addition, this is helpful when this nodes doesn't have DNS or network set up correctly, and it's still able to show the helper method.